src/backend/fts/ftsmessagehandler.c

```diff
@@ -313,13 +313,15 @@ HandleFtsWalRepSyncRepOff(void)
 		false, /* RequestRetry */
 	};
 
+#ifdef FAULT_INJECTOR
 	if (FaultInjector_InjectFaultIfSet("fts_probe",
 										   DDLNotSpecified,
 										   "" /* databaseName */,
 										   "" /* tableName */) == FaultInjectorTypeSkip)
 	{
 		SendFtsResponse(&response, FTS_MSG_SYNCREP_OFF);
```

Comment:
The skip logic here seems wrong. Is it really useful?

Reply:
this routine is added when cloudberry codecase init.

Reply:
> The skip logic here seems wrong. Is it really useful?

I have the same question.

The fault injection just call `SendFtsResponse()` without set response's field, it's not clear the usage.
GPDB's `fts_probe` at `skip` operation is used to skip fts in `FtsLoop()`,  but what's the case we get here?

I doubt if there is a real case triggers this line, but even it's triggered by some sql files and get here, the codes will go down and call `SendFtsResponse()` at line 331, but `SendFtsResponse` has already flushed buffer and end current command, a second call will make  it get trouble then.

Reply:
yeah, maybe we should return at line 323?
there is no commit message to help use know more about this.

Reply:
The current logic should be used to confirm that `FaultInjector_InjectFaultIfSet("fts_probe") == skip` is successful? cause FTS bgworker won't send the SYNC_OFF message when we skip the `fts_probe`?